### DISCUSSION TITLE
OpenSource InstaCalc

### DISCUSSION THREAD

**Discussion by vojgin** [kazad/instacalc#76] [General]
Hello @kazad! I’m impressed with your app and been wondering how it actually works. Do you have any plans on sharing the source code we could explore and learn from?

**Comment by kazad** [comment]
Thanks, glad you're enjoying it! I'd like to open source it as a learning project one day (it needs some cleanups). But if you want to poke around the internals:

* https://instacalc.com/src/grammar/ - runs a test suite, you can browse the parse tree for each test
* https://instacalc.com/src/grammar/ic.ne - [NearleyJS](https://nearley.js.org/) grammar for instacalc, which is compiled into [ic.js](https://instacalc.com/src/grammar/ic.js) 
* https://instacalc.com/src/instacalc.engine.js - old, gnarly calc engine, primarily used for unit and function definitions
* https://instacalc.com/src/instacalc.parser.js - latest calc engine

**Reply by vojgin** [reply]
Thanks @kazad ! I'll peak into it.

**Comment by waradu** [comment]
any updates on this @kazad ? I'd love to see the whole source code. The project is really well done, and I'm genuinely interested in it.
